c10/cuda/CUDACachingAllocator.h

```diff
@@ -125,9 +128,39 @@ struct SegmentInfo {
   int64_t active_size = 0;
   cudaStream_t stream = 0;
   bool is_large = false;
+  MempoolId_t owner_private_pool_id = {0, 0};
   std::vector<BlockInfo> blocks;
 };
 
+struct BlockState {
+  int device;
+  cudaStream_t stream;
+  stream_set stream_uses;
+  size_t size;
+  void* ptr;
+  bool allocated;
+  int gc_count;
```

Comment:
I'd suggest also default initializing these fields as all, because you are default constructing this object and the primitive types don't default initialize to zero.